### DISCUSSION TITLE
Best way to fix incorrect values? updated() is not helping me.

### DISCUSSION THREAD

**Discussion by elfeffe** [livewire/livewire#4621] [1. Help]
I need to avoid users writing 0 in quantity for an item's table.
Validation can help, but I want just to replace 0 with 1.

If I control this on updated(), I use ray() to debug the items array and I see the quantity was changed to 1, which is ok.
But the component's HTML still shows 0.

Where is the best way to make it work?
I have tried hydrate(), and it also doesn't help me.

**Comment by silnarm** [comment]
updated() or upatedFooBar() are where you want to do this.
Changes made there will 'stick', if you're finding it is reset to 0 when the render() comes around, then that is happening somewhere else, and will surely still be happening if you remove the updated() and try to set it to 1 some other way. You need to figure out where/how the reset to 0 is happening, not find an alternative location to set your 1, you are doing that in the correct place.

**Comment by joshhanley** [comment]
@elfeffe what @silnarm has said is correct, something else must be causing issues with data in the view. Sharing your component class and view would help so we can have a look at it 🙂

**Comment by elfeffe** [comment]
It's hard to share the component because I use several Traits.
updated() is running inside one of the Traits, I believe that's the problem.
If I create a custom function() and I call it at render() in the main component, it works.

**Reply by joshhanley** [reply]
@elfeffe hard to help without being able to see some code though. Can you try reproducing the error in a new example component that has one trait that demonstrates it so we can investigate?
